ec/ecdatasource/deploymenttemplates/schema.go

```diff
@@ -0,0 +1,173 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymenttemplates
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (d *DataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve a list of available deployment templates.",
+		Attributes: map[string]schema.Attribute{
+			"region": schema.StringAttribute{
+				Description: "Region to select. For Elastic Cloud Enterprise (ECE) installations, use `ece-region`.",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "Filters for a deployment template with this id.",
+				Optional:    true,
+			},
+			"stack_version": schema.StringAttribute{
```

Comment:
Does it make sense to allow using `stack_version` or `show_deprecated` if `id` is specified?

Reply:
No, best to add a validator that id makes no sense together with the other two options.